modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -947,8 +949,7 @@ def _get_dict_of_block_index(self, axis, indices):
         if isinstance(indices, slice) and (
             indices.step is not None and indices.step != 1
         ):
-            list_of_indices = range(*indices.indices(len(self.axes[axis])))
-            return self._get_dict_of_block_index(axis=axis, indices=list_of_indices)
```

Comment:
why `return` was removed?

Reply:
I just removed a recursive call of `_get_dict_of_block_index`, it appeared, that there is no sense in computing a new `indexer` parameter and calling the function again if we can just process a non-one-step indexer here and continue the execution